### PR TITLE
fix load local

### PR DESCRIPTION
**Pull request by mooleetzi**
## What type of PR is this?

- [ ] API-change
- [X] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #7814 

## What this PR does / why we need it:
1. use closed pipe to notify load local goroutine inner error
2. fix build load about load local
3. add comment

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/plan/build_load.go [hunk 1]
```diff
@@ -133,10 +133,10 @@ func buildLoad(stmt *tree.Load, ctx CompilerContext) (*Plan, error) {
 }
 
 func checkFileExist(param *tree.ExternParam) error {
-	param.Ctx = context.TODO()
 	if param.Local {
 		return nil
 	}
+	param.Ctx = context.TODO()
```
Comment (daviszhen) [inline pkg/sql/plan/build_load.go:139]: param.Ctx uses the requestCtx instead of context.TODO

File: pkg/frontend/mysql_cmd_executor.go [hunk 2]
```diff
@@ -3327,11 +3330,12 @@ func (mce *MysqlCmdExecutor) processLoadLocal(ctx context.Context, param *tree.E
 	if length == 0 {
 		return
 	}
+	quickWrite := false
```
Comment (daviszhen) [inline pkg/frontend/mysql_cmd_executor.go:3333]: quickWrite is confusing. What does it works?

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/sql/plan/build_load.go (hunk 1 above)
Comment (mooleetzi) [reply pkg/sql/plan/build_load.go]: done

File: pkg/frontend/mysql_cmd_executor.go (hunk 2 above)
Comment (mooleetzi) [reply pkg/frontend/mysql_cmd_executor.go]: added some comment in code

**Review by daviszhen — APPROVED** [review]

**Pull request merged**